test/rubygems/test_gem_command_manager.rb

```diff
@@ -49,13 +49,26 @@ def test_find_command_ambiguous_exact
   end
 
   def test_find_command_unknown
-    e = assert_raises Gem::CommandLineError do
+    e = assert_raises Gem::UnknownCommandError do
       @command_manager.find_command 'xyz'
     end
 
     assert_equal 'Unknown command xyz', e.message
   end
 
+  def test_find_command_unknown_suggestions
+    e = assert_raises Gem::UnknownCommandError do
+      @command_manager.find_command 'pish'
+    end
+
+    message = 'Unknown command pish'.dup
+    if Gem::UnknownCommandError.instance_methods.include?(:corrections)
+      message << "\nDid you mean?  \"push\""
+    end
```

Comment:
Something seems to be misbehaving here:

```
  1) Failure:
TestGemCommandManager#test_find_command_unknown_suggestions [/home/travis/build/rubygems/rubygems/test/rubygems/test_gem_command_manager.rb:71]:
--- expected
+++ actual
@@ -1 +1,2 @@
-"Unknown command pish"
+"Unknown command pish
+Did you mean?  \"push\""

```

([link](https://travis-ci.org/rubygems/rubygems/jobs/565517711#L285)